07-MemoryManagementUserSpace/main.c

```diff
@@ -0,0 +1,68 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <alloca.h>
+#include <time.h>
+#include <math.h>
+
+struct timespec get_timestamp(void)
+{
+	struct timespec ts;
+	timespec_get(&ts, TIME_UTC);
+	return ts;
+}
+
+struct timespec sub(struct timespec arg1, struct timespec arg2)
+{
+	struct timespec ret;
+	ret.tv_sec = arg1.tv_sec - arg2.tv_sec;
+	ret.tv_nsec = arg1.tv_nsec - arg2.tv_nsec;
+	if(ret.tv_nsec < 0)
+	{
+		ret.tv_nsec += 1000000000;
+		ret.tv_sec -= 1;
+	}
+	return ret;
+}
+
+const int testsNum = 3;
+const char *label[] = {"malloc", "calloc", "alloca"};
+void *ptr;
```

Comment:
Is it required to make ptr static?

Reply:
Yes, it must be static.

Reply:
Is it static in the last version of your code?